### ISSUE TITLE
Drawing lines and rectangles

### ISSUE THREAD

**Issue by FraserElectronics** [jingwood/d2dlib#107]
I would like to draw a one pixel width rectangle on my D2DControl surface using DrawRectangle with the default stroke width of 1.
However, the rectangle is two pixels wide on my monitor and seems to be a blend of the foreground and background colours.

![5](https://user-images.githubusercontent.com/19693425/217100567-e6003f36-b896-44f9-ba8b-fbb26209e22f.png)

The line should be Red i.e. R:255,G:0,B:0 but you can see it is not.
Two questions:

1. How to I get a single pixel line (is this related to DPI as per my previous issues)?
2. How to *not* blend the pen colour with the background colour?

Andy

**Comment by drewnoakes** [comment]
This Stack Overflow question might be helpful: https://stackoverflow.com/questions/10763762/pixels-in-direct2d

**Comment by FraserElectronics** [comment]
Yes, I suspected that this might be the case - thanks for the link 👍

Andy

**Comment by FraserElectronics** [comment]
@jingwood Are you saying that even if I configure my application to support high DPI I will still have a problem because of a bug in d2dlib?

Andy

**Comment by FraserElectronics** [comment]
Actually, I can answer that as I have followed the steps you link and it has made no change to the rendering on the D2DControl :-(

Andy

**Comment by jingwood** [comment]
>  Are you saying that even if I configure my application to support high DPI I will still have a problem because of a bug in d2dlib?
Yes, but it's another problem.

For the antialias, could you please post an image that contains a normal .NET control how to be rendered like my above?

**Comment by FraserElectronics** [comment]
Do you mean something like this?

![6](https://user-images.githubusercontent.com/19693425/217569125-459edc19-b6d4-45ad-8fc2-c6274de639ac.png)

Andy

**Comment by FraserElectronics** [comment]
This is what it looks like if I remove all the suggestions from the link you mentioned.

![7](https://user-images.githubusercontent.com/19693425/217569889-0835cd3b-79c3-465d-b7e6-dba5a0549c09.png)

Andy
